app/database/admin/database_manager/index.ts

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {MIGRATION_EVENTS, MM_TABLES} from '@constants/database';
+import {Database, Model, Q} from '@nozbe/watermelondb';
+import SQLiteAdapter from '@nozbe/watermelondb/adapters/sqlite';
+import {Class} from '@nozbe/watermelondb/utils/common';
+import type {DefaultNewServer, MigrationEvents, MMDatabaseConnection} from '@typings/database/database';
+import IServers from '@typings/database/servers';
+import {deleteIOSDatabase, getIOSAppGroupDetails} from '@utils/mattermost_managed';
+import {DeviceEventEmitter, Platform} from 'react-native';
+
+import DefaultMigration from '../../default/migration';
+import {App, Global, Servers} from '../../default/models';
+import {defaultSchema} from '../../default/schema';
+import ServerMigration from '../../server/migration';
+import {
+    Channel,
+    ChannelInfo,
+    ChannelMembership,
+    CustomEmoji,
+    Draft,
+    File,
+    Group,
+    GroupMembership,
+    GroupsInChannel,
+    GroupsInTeam,
+    MyChannel,
+    MyChannelSettings,
+    MyTeam,
+    Post,
+    PostMetadata,
+    PostsInChannel,
+    PostsInThread,
+    Preference,
+    Reaction,
+    Role,
+    SlashCommand,
+    System,
+    Team,
+    TeamChannelHistory,
+    TeamMembership,
+    TeamSearchHistory,
+    TermsOfService,
+    User,
+} from '../../server/models';
+import {serverSchema} from '../../server/schema';
+
+// TODO [x] : Initialize a db connection with default schema
+// TODO [x] : handle migration
+// TODO [x] : create server db
+// TODO [x] : set active db
+// TODO [x] : delete server db and removes its record in default db
+
+// TODO [] : retrieve all dbs or a subset via the url and it then returns db instances
+
+// TODO [] : factory reset - wipe every data on the phone
+
+// TODO [] : how do we track down if migration succeeded on all the instances of 'server db'
+
+// TODO :  review all private/public methods/fields
+
+type Models = Class<Model>[]
+
+// A database connection is of type 'Database'; unless it fails to be initialize and in which case it becomes 'undefined'
+type DBInstance = Database | undefined
+
+// The elements needed to create a new connection
+type DatabaseConnection = { databaseConnection: MMDatabaseConnection, shouldAddToDefaultDB: boolean }
+
+// The elements required to switch to another active server database
+type ActiveServerDatabase = { displayName: string, serverUrl: string }
+
+// The elements required to delete a database on iOS
+type RemoveDatabaseIOS = { databaseName?: string, shouldRemoveDirectory?: boolean}
+
+// The only two types of databases in the app
+export enum DatabaseType {
+    DEFAULT,
+    SERVER
+}
+
+class DatabaseManager {
```

Comment:
It would be really good to define an interface for this class; the interface can then act as the "to-do" list and it's clear to see what's still left to do :)

Reply:
I will do it for my next class